cookbook-react/src/recipes/screens/registration/index.tsx

```diff
@@ -4,76 +4,116 @@ import i18next from 'i18next';
-          name={FIELDS.name}
+          className={`${styles.inputContainer} item-1 m-right-2`}
+          label={i18next.t('Registration:firstName')}
+          name={FIELDS.firstName}
           inputType="text"
           inputClassName={styles.input}
-          placeholder={i18next.t('Registration:namePlaceholder') as string}
-          onChange={handleInputChange}
+          placeholder={i18next.t('Registration:firstNamePlaceholder') as string}
+          inputRef={register(VALIDATIONS.firstName)}
+          error={errors?.firstName?.message}
         />
         <FormInput
-          className="full-width m-left-1"
-          label={i18next.t('Registration:surname')}
-          name={FIELDS.surname}
+          className={`${styles.inputContainer} item-1`}
+          label={i18next.t('Registration:lastName')}
+          name={FIELDS.lastName}
           inputType="text"
           inputClassName={styles.input}
-          placeholder={i18next.t('Registration:surnamePlaceholder') as string}
-          onChange={handleInputChange}
+          placeholder={i18next.t('Registration:lastNamePlaceholder') as string}
+          inputRef={register(VALIDATIONS.lastName)}
+          error={errors?.lastName?.message}
         />
       </div>
       <div className={`row ${styles.sectionContainer}`}>
         <FormInput
-          className={styles.inputContainer}
+          className={`${styles.inputContainer} item-1`}
           label={i18next.t('Registration:email')}
           name={FIELDS.email}
-          inputType="text"
+          inputType="email"
           inputClassName={styles.input}
           placeholder={i18next.t('Registration:emailPlaceholder') as string}
-          onChange={handleInputChange}
+          inputRef={register(VALIDATIONS.email)}
+          error={errors?.email?.message}
         />
       </div>
       <div className={`row space-between ${styles.sectionContainer}`}>
         <FormInput
-          className={styles.inputContainer}
+          className={`${styles.inputContainer} item-1 m-right-2`}
           label={i18next.t('Registration:password')}
           name={FIELDS.password}
           inputType="password"
           inputClassName={styles.input}
           placeholder={i18next.t('Registration:passwordPlaceholder') as string}
-          onChange={handleInputChange}
+          inputRef={register(VALIDATIONS.password)}
+          error={errors?.password?.message}
         />
         <FormInput
-          className="full-width m-left-1"
+          className={`${styles.inputContainer} item-1`}
           label={i18next.t('Registration:confirmPassword')}
           name={FIELDS.confirmPassword}
           inputType="password"
           inputClassName={styles.input}
           placeholder={i18next.t('Registration:confirmPasswordPlaceholder') as string}
-          onChange={handleInputChange}
+          inputRef={register({ ...VALIDATIONS.confirmPassword, validate: validateConfirmPassword })}
```

Comment:
Why isn't `validateConfirmPassword', part of `VALIDATIONS.confirmPassword`?

Reply:
because it uses `watch`, which is returned by the useForm hook from RHF.